code/modules/RCD/RPD.dm

```diff
@@ -136,6 +139,103 @@
 	if (selected)
 		return selected.Topic(href, href_list)
 
+/obj/item/device/rcd/rpd/slime_act(primarytype, mob/user)
+	if(primarytype == /mob/living/carbon/slime/metal)
+		if(has_metal_slime)
+			to_chat(user, "It already has a slime extract attached.")
+			return FALSE
+		else
+			has_metal_slime=1
+			verbs += /obj/item/device/rcd/rpd/proc/multilayer
+			to_chat(user, "You jam the slime extract into the RPD's fabricator.")
+			return TRUE
+
+	if(primarytype == /mob/living/carbon/slime/yellow)
+		if(has_yellow_slime)
+			to_chat(user, "It already has a slime extract attached.")
+			return FALSE
+		else
+			has_yellow_slime=1
+			verbs += /obj/item/device/rcd/rpd/proc/autowrench
+			to_chat(user, "You jam the slime extract into the RPD's output nozzle.")
+			return TRUE
+
+/obj/item/device/rcd/rpd/afterattack(var/atom/A, var/mob/user)
+	if(!selected)
+		return 1
+
+	if(~selected.flags & (RCD_SELF_SANE | RCD_RANGE) && !(user.Adjacent(A) && A.Adjacent(user))) // If RCD_SELF_SANE and RCD_RANGE are disabled we use adjacency.
+		return 1
+
+	if(selected.flags & RCD_RANGE && ~selected.flags & RCD_SELF_SANE && get_dist(A, user) > 1) // RCD_RANGE is used AND we're NOT SELF_SANE, use range(1)
+		return 1
+
+	if(selected.flags & RCD_GET_TURF) // Get the turf because RCD_GET_TURF is on.
+		A = get_turf(A)
+		if (!A)
+			return // Thing clicked was in nullspace, so we won't pass a null turf.
+
+	if(~selected.flags & RCD_SELF_SANE && get_energy(user) < selected.energy_cost) // Handle energy amounts, but only if not SELF_SANE.
+		return 1
+	if(build_all && istype(selected, /datum/rcd_schematic/pipe))
+		var/datum/rcd_schematic/pipe/our_schematic = selected //typecast
+		if(our_schematic.layer) // this is needed because disposal pipe schematic datums are retarded
+			for(var/layer in 1 to 5)
+				busy  = TRUE // Busy to prevent switching schematic while it's in use.
+				our_schematic.set_layer(layer)
+				if(layer != 5 )
+					spawn(-1) our_schematic.attack(A, user)
```

Comment:
no one line spawn, also why -1?

Reply:
I fixed that one line, also not sure, didn't code the -1 so don't know the significance.